RetakesAllocator/RoundType.cs

```diff
@@ -1,8 +1,84 @@
-﻿namespace RetakesAllocator;
+﻿using CounterStrikeSharp.API.Modules.Entities.Constants;
+using CounterStrikeSharp.API.Modules.Utils;
+
+namespace RetakesAllocator;
 
 public enum RoundType
 {
     Pistol,
     HalfBuy,
     FullBuy,
 }
+
+public static class RoundTypeHelpers
+{
+    public static RoundType GetRandomRoundType()
+    {
+        var randomValue = new Random().NextDouble();
+
+        return randomValue switch
+        {
+            // 15% chance of pistol round
+            < 0.15 => RoundType.Pistol,
+            // 25% chance of halfbuy round
+            < 0.40 => RoundType.HalfBuy,
+            // 60% chance of fullbuy round
+            _ => RoundType.FullBuy,
+        };
+    }
+
+    public static IEnumerable<CsItem> GetRandomUtilForRoundType(RoundType roundType, CsTeam team)
+    {
+        // Limited util on pistol rounds
+        if (roundType == RoundType.Pistol)
+        {
+            return new List<CsItem>
+            {
+                Utils.Choice(new List<CsItem>
+                {
+                    CsItem.Flashbang,
+                    CsItem.Smoke,
+                }),
+            };
+        }
+
+        // All util options are available on buy rounds
+        var possibleItems = new List<CsItem>
+        {
+            CsItem.Flashbang,
+            CsItem.Smoke,
+            CsItem.HEGrenade,
+            team == CsTeam.Terrorist ? CsItem.Molotov : CsItem.Incendiary,
+        };
+
+        // Everyone gets one util
+        var randomUtil = new List<CsItem>
+        {
+            Utils.Choice(possibleItems),
+        };
```

Comment:
Store the first grenade they get allocated. Then if that is != a flashbang remove it from the possibleItems list. and you can just run the same allocation.